### ISSUE TITLE
Microstuttering with "poor network connection" on Android

### ISSUE THREAD

**Issue by MagCat** [moonlight-stream/moonlight-android#328]
Hi, I have been having this problem where my Samsung S7 Verizon having 'poor network connection' message pops up whenever I have microstutter really badly (like every few 3-5 seconds). What I used to solve was doing factory reset on my phone, but every time I do that the problem keeps coming back randomly few days or weeks later. It's really annoying I have to reset everything to make it work, so I want to know if the devs have known this issue and looked into this.

My PC specs is:
i7 4790k 4.0 GHz 
GTX 970 
16gb DDR3 RAM

I have always used 5.0 Ghz channel on my router and please note that it always works perfectly up until I do something to the bluetooth or whatever that caused this issue.  I used to have the app working perfectly until the problem comes back in next morning, and once the microstutter happened, it keeps happening every time until I do another factory reset. Please note that any setting (720p 30 fps in 1 Mbps or forced H.265 setting) won't even help. Everytime I exit the game I notice the latency is only 8-9ms or 14-18ms at most, so the microstuttering doesn't make sense because my network is totally perfect. Since I had this problem for **MONTHS**, I deduced that something has been in conflict with my phone's OS, and I need to know what can be fixed. My phone is currently rooted, but the problem happened all the times even before I get it rooted.

**Comment by genmatrix** [comment]
Can you post a video and a logcat? I want to compare against my similar issues on my Pixel C.

**Comment by MagCat** [comment]
I can't capture video without my phone. I can only say that it freezes every few 3-5 seconds (the freezes can stay as bad as 2 seconds) and being unplayable. I will try to get logcat later in few more hours.

**Comment by genmatrix** [comment]
I use AZ Screen Recorder app. Not sure if it will work with your device. It's a free app too.

**Comment by MagCat** [comment]
I'm still trying to figure how to start logcat, but I got the video uploaded here.
https://www.youtube.com/watch?v=Q3a1Q866CXU
The video shows some necessary information about my specs and GFE version I used as well, along with the config I use for moonlight. Though I used the same config just fine a day before, if not even with more bit rate.

**Comment by genmatrix** [comment]
Heres a link that walks you through how to do it. Since you're already rooted, it shouldn't be too bad. 
https://android.stackexchange.com/questions/33216/how-can-i-get-a-logcat

**Comment by MagCat** [comment]
Sorry for late reply. I had other business going on. I tried LogCat app but didn't know how to copypaste the whole log, so I had to go back to traditional way. Any case, 
[catlog.txt](https://github.com/moonlight-stream/moonlight-android/files/720937/catlog.txt)
I attached the log file here.

**Comment by MagCat** [comment]
Looks like no one knows other ways to fix this problem. I assume that moonlight just really hates Marshmallow devices.

**Comment by genmatrix** [comment]
You try an Ethernet adapter yet? It could just be something wifi. The dev does all of this on his spare time so don't get discouraged.

**Comment by MagCat** [comment]
I haven't tried with Ethernet adapter yet, but I looked up and it doesn't seem like I can use it for my phone, and my PC already used ethernet cable to my router and modem. I know devs has been doing this for his spare time, but a problem is still a problem. I don't blame anyone here.

**Comment by genmatrix** [comment]
Right, right. This may be device specific but have you turned off wifi scanning under location services? Also, when loading moonlight have you completely closed out of the settings app?

I've found it doesn't give me the poor connection message when I make sure the Settings app isn't loaded in the background. I guess my problem is that it's scanning for other wifi access points if it's left open in the background and causing my lag. 

Your stuttering is a bit different than mine (https://www.youtube.com/watch?v=c7tfKtHOUu8) but it's something to try.

**Comment by MagCat** [comment]
I did turn off Location service off. I used to have my LG G4 before and I had no problem playing moonlight for entire year, but that was on 5.0 Android. I also tried to make sure nothing else running besides moonlight but it gave me no different result. I also watched your video and noticed the message pops up at 0:53, so it looks like you have similar issues to mine actually. However, your guess can be something I can try next time, since I reset my phone to factory setting again.
